confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java

```diff
@@ -590,7 +591,54 @@ public int getUnknownDataNodesNum() {
     return allDataNodes.size();
   }
 
-  public void addMetrics() {
+  /**
+   * Get the LeaderCount of each DataNodeId
+   *
+   * @return Map<DataNodeId, LeaderCount>
+   */
+  public Map<Integer, Integer> getLeadershipCountByDatanode() {
+    Map<Integer, Integer> idToCountMap = new ConcurrentHashMap<>();
+
+    regionGroupCacheMap.forEach(
```

Comment:
602-610 can be replaced with `regionGroupCacheMap.forEach(
        (consensusGroupId, regionGroupCache) -> idToCountMap.merge(regionGroupCache.getLeaderDataNodeId(), 1, Integer::sum));`

Reply:
Okay, learned ~